api/v1alpha1/servicedefaults_types_test.go

```diff
@@ -654,3 +654,88 @@ func TestMatchesConsul(t *testing.T) {
 		})
 	}
 }
+
+func TestValidate(t *testing.T) {
+	cases := map[string]struct {
+		input          *ServiceDefaults
+		expectedErrMsg string
+	}{
+		"meshgateway.mode": {
+			&ServiceDefaults{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "my-service",
+				},
+				Spec: ServiceDefaultsSpec{
+					MeshGateway: MeshGatewayConfig{
+						Mode: "foobar",
+					},
+				},
+			},
+			`ServiceDefaults.consul.hashicorp.com "my-service" is invalid: spec.meshGateway.mode: Invalid value: "foobar": must be one of "remote", "local", "none" or ""`,
```

Comment:
Is it normal for this to be camelcased? (ServiceDefauts.consul.hashicorp.com)

Reply:
I used the reference from the kubebuilder book. It does look kinda ugly though. Ill switch it to being kebab case.